app/ytdl.py

```diff
@@ -113,14 +118,19 @@ async def update_status(self):
             if status is None:
                 return
             self.tmpfilename = status.get('tmpfilename')
+            if 'filename' in status:
+                self.info.filename = os.path.basename(status.get('filename'))
             self.info.status = status['status']
-            self.info.msg = status.get('msg')
+            if 'msg' in status:

```

Comment:
why do you need the condition? status.get('msg') will return None if 'msg' doesn't exist in status.
(same goes for speed and eta below)

Reply:
I thought a 'None' would lead to problems in the UI, because the filename is always sent after download has finished (filename is from the postprocessing step), so the former "good" values of 'msg', 'speed', and 'eta' would be overwritten with 'None'. However, at first glance, it does not cause issues; 'speed' and 'eta' are not required anymore after download has finished.

Reply:
I see your point, it makes sense. So yeah, speed and eta aren't required, but maybe it's worth keeping msg after all? I haven't thought of the case you're talking about (values being rewritten to None when the download finishes).

Reply:
What does msg contain? If it's only an error message if download/conversion/... fails, then it shouldn't be overwritten since postprocessing should only occur if previous steps are successful.